go/vt/mysqlctl/clone.go

```diff
@@ -0,0 +1,330 @@
+func (c *CloneExecutor) validateDonorRemote(ctx context.Context) error {
+	params := &mysql.ConnParams{
+		Host:  c.DonorHost,
+		Port:  c.DonorPort,
+		Uname: c.DonorUser,
+		Pass:  c.DonorPassword,
+	}
+
+	conn, err := mysql.Connect(ctx, params)
+	if err != nil {
+		return fmt.Errorf("failed to connect to donor %s:%d: %w", c.DonorHost, c.DonorPort, err)
+	}
+	defer conn.Close()
+
+	// Check MySQL version
+	qr, err := conn.ExecuteFetch("SELECT @@version", 1, false)
+	if err != nil {
+		return fmt.Errorf("failed to query donor MySQL version: %w", err)
+	}
+	if len(qr.Rows) == 0 || len(qr.Rows[0]) == 0 {
+		return errors.New("empty version result from donor")
+	}
+	versionStr := qr.Rows[0][0].ToString()
+	capableOf := mysql.ServerVersionCapableOf(versionStr)
+	if capableOf == nil {
+		return fmt.Errorf("unable to determine MySQL capabilities for donor version %q", versionStr)
+	}
+	ok, err := capableOf(capabilities.MySQLClonePluginFlavorCapability)
+	if err != nil {
+		return fmt.Errorf("failed to check donor clone capability: %w", err)
+	}
+	if !ok {
+		return fmt.Errorf("donor MySQL CLONE requires version 8.0.17 or higher, got %s", versionStr)
+	}
+
+	// Check clone plugin is installed
+	qr, err = conn.ExecuteFetch("SELECT PLUGIN_STATUS FROM information_schema.PLUGINS WHERE PLUGIN_NAME = 'clone'", 1, false)
+	if err != nil {
+		return fmt.Errorf("failed to check donor clone plugin status: %w", err)
+	}
+	if len(qr.Rows) == 0 {
+		return errors.New("clone plugin is not installed on donor (add 'plugin-load-add=mysql_clone.so' to my.cnf)")
+	}
+	status := qr.Rows[0][0].ToString()
+	if status != "ACTIVE" {
+		return fmt.Errorf("clone plugin is not active on donor (status: %s)", status)
+	}
+
+	// Check for non-InnoDB tables
+	qr, err = conn.ExecuteFetch(`
+		SELECT TABLE_SCHEMA, TABLE_NAME, ENGINE
+		FROM information_schema.TABLES
+		WHERE ENGINE != 'InnoDB'
+		AND ENGINE IS NOT NULL
+		AND TABLE_TYPE = 'BASE TABLE'
+		AND TABLE_SCHEMA NOT IN ('mysql', 'information_schema', 'performance_schema', 'sys')
+	`, 1000, false)
+	if err != nil {
+		return fmt.Errorf("failed to check donor for non-InnoDB tables: %w", err)
+	}
+	if len(qr.Rows) > 0 {
+		var tables []string
+		for _, row := range qr.Rows {
+			schema := row[0].ToString()
+			table := row[1].ToString()
+			engine := row[2].ToString()
+			tables = append(tables, fmt.Sprintf("%s.%s (%s)", schema, table, engine))
+		}
+		return fmt.Errorf("non-InnoDB tables found on donor (CLONE only supports InnoDB): %s", strings.Join(tables, ", "))
+	}
+
+	log.Infof("Donor %s:%d validated successfully (MySQL %s)", c.DonorHost, c.DonorPort, versionStr)
+	return nil
+}
+
+// checkCloneCapability verifies that the MySQL version supports the CLONE plugin.
+func (c *CloneExecutor) checkCloneCapability(ctx context.Context, mysqld MysqlDaemon) error {
+	result, err := mysqld.FetchSuperQuery(ctx, "SELECT @@version")
+	if err != nil {
+		return fmt.Errorf("failed to query MySQL version: %w", err)
+	}
```

Comment:
Optional: we're grabbing `@@version` more than once, once here as a super user, once as a regular user. Maybe we should pass one result around?